x-pack/platform/plugins/shared/stack_connectors/server/connector_types/openai/openai.ts

```diff
@@ -76,28 +95,95 @@ export class OpenAIConnector extends SubActionConnector<Config, Secrets> {
       ...('projectId' in this.config ? { 'OpenAI-Project': this.config.projectId } : {}),
     };
 
-    this.openAI =
-      this.config.apiProvider === OpenAiProviderType.AzureAi
-        ? new OpenAI({
-            apiKey: this.secrets.apiKey,
-            baseURL: this.config.apiUrl,
-            defaultQuery: { 'api-version': getAzureApiVersionParameter(this.config.apiUrl) },
-            defaultHeaders: {
-              ...this.headers,
-              'api-key': this.secrets.apiKey,
-            },
-          })
-        : new OpenAI({
-            baseURL: removeEndpointFromUrl(this.config.apiUrl),
-            apiKey: this.secrets.apiKey,
-            defaultHeaders: {
-              ...this.headers,
-            },
+    // Assign configAny for dynamic property access
+    this.configAny = this.config as Config & {
+      certificateFile?: string | string[];
+      certificateData?: string;
+      privateKeyFile?: string | string[];
+      privateKeyData?: string;
+      caFile?: string | string[];
+      caData?: string;
+      verificationMode?: 'full' | 'certificate' | 'none';
+    };
+    this.logger.debug(
+      `Provider: ${this.provider}, certificateFile: ${this.configAny.certificateFile}, certificateData: ${this.configAny.certificateData}, privateKeyFile: ${this.configAny.privateKeyFile}, privateKeyData: ${this.configAny.privateKeyData}, caFile: ${this.configAny.caFile}, caData: ${this.configAny.caData}, verificationMode: ${this.configAny.verificationMode}`
+    );
+
+    try {
+      if (
+        this.provider === OpenAiProviderType.Other &&
+        (this.configAny.certificateFile ||
+          this.configAny.certificateData ||
+          this.configAny.privateKeyFile ||
+          this.configAny.privateKeyData)
+      ) {
+        // Validate PKI configuration
+        if (
+          !validatePKICertificates(
+            this.logger,
+            this.configAny.certificateFile,
+            this.configAny.certificateData,
+            this.configAny.privateKeyFile,
+            this.configAny.privateKeyData
+          )
+        ) {
+          throw new Error('Invalid or inaccessible PKI certificates');
+        }
+        const sslOverrides = getPKISSLOverrides(
```

Comment:
this is a lot of arguments, please refactor to take an object so that arguments out of order do not matter